### PR TITLE
rec: Prep for rec-4.5.0-alpha3

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Including a few upgrade notes.

https://builder.powerdns.com/#/builders/97/builds/1704

blog post:
https://blog.powerdns.com/?p=3317&preview=true

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/docs/changelog/4.5.rst [hunk 1]
```diff
@@ -1,6 +1,126 @@
...
+  .. change::
+    :tags: Bug Fixes
+    :pullreq: 10111
+    :tickets: 10080
+
+    Handle policy (if needed) after postresolve and document the hooks better.
+
+  .. change::
+    :tags: Improvements
+    :pullreq: 9361
```
Comment (rgacogne) [inline pdns/recursordist/docs/changelog/4.5.rst:41]: I'm pretty sure that PR impacted only dnsdist?

File: pdns/recursordist/docs/changelog/4.5.rst [hunk 2]
```diff
@@ -1,6 +1,126 @@
...
+    :pullreq: 10089
+    :tickets: 10058
+
+    Count DNSSEC stats for given names in a different set of counters.
+
+  .. change::
+    :tags: Improvements
+    :pullreq: 10096
+
+    Remember non resolving ns.
```
Comment (rgacogne) [inline pdns/recursordist/docs/changelog/4.5.rst:70]:
```suggestion
    Remember non resolving nameservers.
```

**Review by rgacogne — CHANGES_REQUESTED** [review]

File: pdns/recursordist/docs/changelog/4.5.rst (hunk 1 above)
Comment (Habbie) [reply pdns/recursordist/docs/changelog/4.5.rst:41]: indeed

File: pdns/recursordist/docs/upgrade.rst [hunk 3]
```diff
@@ -26,12 +26,19 @@ The next release will start deprecating them.
 Users are advised to start using the new names to avoid future
 trouble.
 
+Special Domains
+^^^^^^^^^^^^^^^
+Queries for all names in the ``.localhost`` domain will answer in accordance with :rfc:`6761` section 6.3 point 4.
+That means that they will be answered with ``127.0.0.1``, ``::1` or a negative` response.
```
Comment (mnordhoff) [inline pdns/recursordist/docs/upgrade.rst:32]:
```suggestion
That means that they will be answered with ``127.0.0.1``, ``::1`` or a negative response.
```
Backtick moved around.

**Review by pieterlexis — APPROVED** [review]

**Pull request merged**